python/ctsm/subset_data.py

```diff
@@ -0,0 +1,614 @@
+        required=False,
+        default=False,
+    )
+    # -- region-specific parser options
+    rg_parser.add_argument(
+        "--lat1",
+        help="Region start latitude. [default: %(default)s]",
+        action="store",
+        dest="lat1",
+        required=False,
+        type=plat_type,
+        default=-40,
+    )
+    rg_parser.add_argument(
+        "--lat2",
+        help="Region end latitude. [default: %(default)s]",
+        action="store",
+        dest="lat2",
+        required=False,
+        type=plat_type,
+        default=15,
+    )
+    rg_parser.add_argument(
+        "--lon1",
+        help="Region start longitude. [default: %(default)s]",
+        action="store",
+        dest="lon1",
+        required=False,
+        type=plon_type,
+        default=275.0,
+    )
+    rg_parser.add_argument(
+        "--lon2",
+        help="Region end longitude. [default: %(default)s]",
+        action="store",
+        dest="lon2",
+        required=False,
+        type=plon_type,
+        default=330.0,
+    )
+    rg_parser.add_argument(
+        "--reg",
+        help="Region name or tag. [default: %(default)s]",
+        action="store",
+        dest="reg_name",
+        required=False,
+        type=str,
+        default="",
+    )
+    rg_parser.add_argument(
+        "--create-mesh",
+        help="Subset a mesh file for a region.",
+        action="store_true",
+        dest="create_mesh",
+        required=False,
+        default=False,
+    )
+
+    # -- common options between both subparsers
+    for subparser in [pt_parser, rg_parser]:
+        subparser.add_argument(
+            "--create-domain",
+            help="Create CLM domain file at single point/region.",
+            action="store_true",
+            dest="create_domain",
+            required=False,
+            default=False,
+        )
+        subparser.add_argument(
+            "--create-surface",
+            help="Create surface data file at single point/region.",
+            action="store_true",
+            dest="create_surfdata",
+            required=False,
+            default=False,
+        )
+        subparser.add_argument(
+            "--create-landuse",
+            help="Create landuse data file at single point/region.",
+            action="store_true",
+            dest="create_landuse",
+            required=False,
+            default=False,
+        )
+        subparser.add_argument(
+            "--create-datm",
+            help="Create DATM forcing data at single point/region.",
+            action="store_true",
+            dest="create_datm",
+            required=False,
+            default=False,
+        )
+        subparser.add_argument(
+            "--create-user-mods",
+            help="Create user mods directories and files for running CTSM with the subset data.",
+            action="store_true",
+            dest="create_user_mods",
+            required=False,
+            default=True,
```

Comment:
I agree that listing the default is confusing - maybe we can remove?

Reply:
The defaults are removed now for `store_true` and `store_false` flags.

Reply:
But I have to admit I slightly like to specify the defaults explicitly better, especially because we already have the `required=False,` and the user might read that line as the default. But I agree that it is confusing too. 

For this PR we removed the defaults for `store_true` flags. (we don't have any `store_false` args anymore.)

Reply:
FYI: This is addressed and all the defaults are removed for `store_true` flags.

Reply:
I only found one instance where an argument used store_true, but also had a default. That is --create-mesh, in subset_data.py. So one more to handle.

Reply:
GTG